ModeratorFrontEnd/AIForOrcas/AIForOrcas.Server/Helpers/MetadataFilters.cs

```diff
@@ -8,41 +8,57 @@ public static class MetadataFilters
 	{
 		public static int DefaultRecordsPerPage = 5;
 
-		public static void ApplyTimeframeFilter(ref IQueryable<Metadata> queryable, string timeframe)
-		{
-			if (!string.IsNullOrWhiteSpace(timeframe))
-			{
-				timeframe = timeframe.ToLower();
-
-				if (timeframe != "all")
-				{
-					var now = DateTime.Now;
+        public static void ApplyTimeframeFilter(ref IQueryable<Metadata> queryable, string timeframe, DateTime? dateFrom=null, DateTime? dateTo=null)
+        {
+            if (!string.IsNullOrWhiteSpace(timeframe))
+            {
+                timeframe = timeframe.ToLower();
 
-					if (timeframe == "30m")
-						now = now.AddMinutes(-30);
+                if (timeframe == "all")
```

Comment:
I don't think this is going to work correctly. The way it reads now, for example, would be in order to see the 30m timeframe the timeframe would have to be set to both "all" and "30m" which is not possible. Would think the best way for this would be to convert this back to != "all" and make your new "range" one of the cases.

Reply:
Apologies, this issue came to due to refactoring at the end, and not testing all the scenarios. This is now fixed.